File: ovvium-app/src/model/response/LocationResponse.ts
import { ResourceIdResponse } from './ResourceIdResponse';


export class LocationResponse extends ResourceIdResponse {
    customerId: string;
    positionId: number;
    description: string;

    constructor(locationResponse: LocationResponse = {} as LocationResponse) {
        super(locationResponse.id);
        this.customerId = locationResponse.customerId;
        this.positionId = locationResponse.positionId;
        this.description = locationResponse.description;
    }
}

File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/response/location/LocationResponse.java
package com.ovvium.services.web.controller.bff.v1.transfer.response.location;

import com.ovvium.services.model.customer.Location;
import com.ovvium.services.web.controller.bff.v1.transfer.response.common.ResourceIdResponse;
import lombok.Getter;

@Getter
public final class LocationResponse extends ResourceIdResponse {

	private final String zone;
	private final String serialNumber;
	private final int position;
	private final String description;
	private final boolean advancePayment;

	public LocationResponse(Location location) {
		super(location);
		this.zone = location.getZone().getName();
		this.serialNumber = location.getSerialNumber().getValue();
		this.position = location.getPosition();
		this.description = location.getDescription();
		this.advancePayment = location.isAdvancePayment();
	}

}
